Src/Support/Google.Apis.Auth/OAuth2/Responses/TokenResponse.cs

```diff
@@ -94,30 +95,21 @@ public DateTime Issued
 
         // Note: ideally this would be called ShouldRefresh or similar.
         /// <summary>
-        /// Returns <c>true</c> if the token is expired or it's going expire soon.
+        /// Returns <c>true</c> if the token is expired or it's going to expire soon.
         /// </summary>
-        public bool IsExpired(IClock clock)
-        {
-            if (AccessToken == null || !ExpiresInSeconds.HasValue)
-            {
-                return true;
-            }
-
-            return IssuedUtc.AddSeconds(ExpiresInSeconds.Value - TokenRefreshTimeWindowSeconds) <= clock.UtcNow;
-        }
+        public bool IsExpired(IClock clock) =>
+            (AccessToken == null && IdToken == null) ||
+            !ExpiresInSeconds.HasValue ||
```

Comment:
I think this could do with a bit more explanation. We should comment (at *least* internally) that we regard a token response with no access token or ID token to be expired, and also that if there's no expiry seconds set, that that counts as being expired. (That seems a little odd to me - I'd expect a missing lifetime to be "permanently valid" rather than "never valid".)